File: HW_1/hw1_projective.py
import cv2, skimage.data
import numpy as np
import matplotlib.pyplot as plt

img = skimage.data.astronaut()

# 1) Scaling, Python
s = 1.5
S = np.array([[s, 0, 0],
              [0, s, 0],
              [0, 0, 1]])
scaled = cv2.warpPerspective(img, S, (0, 0))
cv2.imwrite('./fig/scaling_astronaut.png', scaled[:, :, ::-1])

# 2) Rotation, Python
theta = np.pi / 6
R = np.array([[np.cos(theta), -np.sin(theta), 0],
              [np.sin(theta),  np.cos(theta), 0],
              [0,              0,             1]])
rotated = cv2.warpPerspective(img, R, (0, 0))
cv2.imwrite('./fig/rotation_astronaut.png', rotated[:, :, ::-1])

# 3) Similarity, Python
s = 3
theta = -np.pi / 3
Sim = np.array([[s*np.cos(theta), -np.sin(theta),   0],
                [np.sin(theta),   s*np.cos(theta),  0],
                [0,                0,               1]])
similarity = cv2.warpPerspective(img, Sim, (0, 0))
cv2.imwrite('./fig/similarity_astronaut.png', similarity[:, :, ::-1])

# 4) Affine, Python
Aff = np.array([[  2, 0.33, -100], 
                [0.2,    1,   50], 
                [  0,    0,    1]]);
affine = cv2.warpPerspective(img, Aff, (0, 0))
cv2.imwrite('./fig/affine_astronaut.png', affine[:, :, ::-1])

File: HW_1/hw1_projective.m
%{
2022094093 Kim Dohoon
Assignment#1
%}

% Load peppers.png
img = imread('peppers.png');

% Load Own Image - Cherry Blossom
%img = imread('./fig/cb.png');

% 1) Scaling, Matlab
s = 1.5;
S = [s 0 0; 
     0 s 0; 
     0 0 1];
scaled = imwarp(img, projective2d(S), 'OutputView', imref2d(size(img)));
imwrite(scaled, './fig/scaling_peppers.png', 'png');
%imwrite(scaled, './fig/scaling_cb.png', 'png')

% 2) Rotation, Matlab
theta = -pi / 6;
R = [cos(theta) -sin(theta) 0; 
     sin(theta) cos(theta)  0; 
     0          0           1];
rotated = imwarp(img, projective2d(R), 'OutputView', imref2d(size(img)));
imwrite(rotated, './fig/rotation_peppers.png', 'png');
%imwrite(rotated, './fig/rotation_cb.png', 'png')

% 3) Similarity, Matlab
s = 3;
theta = pi / 3;
Sim = [s*cos(theta) -sin(theta)  0; 
       sin(theta)   s*cos(theta) 0; 
       0            0            1];
similarity = imwarp(img, projective2d(Sim), 'OutputView', imref2d(size(img)));
imwrite(similarity, './fig/similarity_peppers.png', 'png');
%imwrite(similarity, './fig/similarity_cb.png', 'png');

% 4) Affine, Matlab
Aff = [   2 0.33 0; 
        0.2    1 0; 
       -100   50 1;];
affine = imwarp(img, projective2d(Aff), 'OutputView', imref2d(size(img)));
imwrite(affine, './fig/affine_peppers.png', 'png');
%imwrite(affine, './fig/affine_cb.png', 'png');
